### ISSUE TITLE
find solution for theming

### ISSUE THREAD

**Issue by oroulet** [QTodoTxt/QTodoTxt2#2]
Currently we have a quite harcoded solution. Ideally it should be qml only but things like icons are part of models in qt so I am not sure how to solve that

**Comment by fuchsmich** [comment]
Generally QtQuick Controls 1 (Buttons, Menus, etc...) should be decorated by default suitable to the Desktop Environment. But the colors of the currently used theme can also be accessed in qml (e.g. for styling custom items):
http://doc.qt.io/qt-5/qml-qtquick-systempalette.html

At least in KDE if you choose the right "icon-name" in Actions, the according icon appears in Menus and Toolbars (icons can be browsed with cuttlefish (part of kde-sdk)). In Windows this doesnt work for me. I guess we have to deploy some kind of default icon-theme with the app.

**Comment by oroulet** [comment]
I agree that we should use system theming as much as possible. I use gnome3. And I have absoluetely no idea how to force a QT theme since they removed the qtconfig app in qt5....

**Comment by fuchsmich** [comment]
Could you show me a screenshot, what the app looks like in gnome?

Am 13.04.2017 um 17:15 schrieb oroulet:

**Comment by oroulet** [comment]
![image](https://cloud.githubusercontent.com/assets/2564046/25014922/07eab552-207a-11e7-9b18-d685a3f8af96.png)

The big empty space appearde when you made the new sorting buttons not visible...

**Comment by fuchsmich** [comment]
Its funny how your system behaves differently to my windows and linux system.
There is no empty space here.

**Comment by oroulet** [comment]
looks like css stylesheets are not supported in QML so we cannot reuse the sheets had...

**Comment by fuchsmich** [comment]
Does this help? http://doc.qt.io/qt-5/qtquick-controls-styles-qmlmodule.html

It looks like the syntax is quite similar.
I will test it later.

Am 14.04.2017 um 10:59 schrieb oroulet:

**Comment by oroulet** [comment]
for information if we use QApplication from QtGui and not QWidgets then we miss looking at standard systems icons (But this is probably a linux only functionality I guess)

**Comment by TRSx80** [comment]
I already made similar comment under #53, but I just wanted to re-state it here. I agree strongly with @oroulet comment of:

> I agree that we should use system theming as much as possible.

FWIW, I am on KDE Plasma with Breeze Dark theme and everything looks very good to me. I know you guys are on different environments, so perhaps I can help the project by reporting what I am seeing in my environment. I don't know enough about coding (yet) but I do want to contribute what I can because I love the project and it meets my needs just about perfectly. :)

**Comment by oroulet** [comment]
If you want to learn coding, this is a good project. The code is simple
with good separation of concern

**Comment by severoraz** [comment]
I am using gnome 3 (3.26) with the dark theme and qtodotxt2 did not adopt the dark colours.

**Comment by oroulet** [comment]
@severoraz I have the same issue. It look like qt5 applcations do not get theme from gnome... While the doc says they should... something does not work....
And it looks like qt5 has no tool to set a theme by hand... any help is welcome on that...
